### PR TITLE
Complete Meeting Scheduling Algorithm MVP

### PR DESCRIPTION
**Pull request by ptwu**
### Summary
This PR implements the meeting scheduling algorithm described in the Week 5 Walkthrough. In this diff,

- Data structures from the `Collection` interface are used to algorithmically solve the calendar scheduling problem.
- Helper methods to help compute the query are written in `FindMeetingQuery.java`
  - Corresponding Javadoc added for these methods
- Test-driven development was used to verify the functionality of the methods 

### Screenshots
N/A

### Test Plan
A test suite has been written already by the STEP program, and it seems to evaluate corner cases and default cases of use for the calendar app. Passing these cases is said to give confidence that the program works, but I will be adding more tests later on this week as well as implementing the optional attendees feature.

Run `mvn test` in `/walkthroughs/week-5-tdd/project` to evaluate the test cases stored in `FindMeetingQueryTest.java`. See that the tests pass.

### Notes
**Comments on Time Complexity:** The time complexity of this algorithm is bounded by `O(mn)`, which is the worst-case for `m` attendees and `n` events for the meeting event. This most expensive operation occurs in the `getBusyTimeRanges` method, where the nested for loop iterates on the attendees list for each event. 

From a practical standpoint, The `sort` method from the `Collections` utility class is used to sort the results from calling the `getBusyTimeRanges` method. It uses the TimSort sorting algorithm known to have worst-case `O(n log n)` runtime. We can reason that this could be greater than `mn` in practice since the # of attendees is likely to be not too large in comparison to the # of events.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,102 @@
...
+      for (String attendee : attendees) {
+        if (currEventAttendees.contains(attendee)) {
+          busyTimeRanges.add(event.getWhen());
+          break;
+        }
+      }
+    }
+    return busyTimeRanges;
+  }
+
```
Comment (chaowangzj) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:46]: In general, put the public methods before private methods. Private methods are usually helper methods, and they are difficult to understand without the context

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,102 @@
...
+      prevTimeRange = curr;
+    }
+  }
+
+  /**
+  * Adds a time range with certain start, end, duration, and inclusitivity attributes to a Collection of 
+  * TimeRange objects if they fit the meeting requirements and are valid within a day.
+  */
+  private void addTimeRangeIfValid(Collection<TimeRange> availableTimeRanges, int start, int end, long duration,
+      boolean inclusive) {
```
Comment (chaowangzj) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:75]: It seems like having too many parameters? Can you create the TimeRange outside the method? It will reduce the # of parameters without changing the purpose of the method.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (ptwu) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:46]: Done.
